python/ray/_private/usage/usage_lib.py

```diff
@@ -182,6 +193,106 @@ def put_cluster_metadata(gcs_client, num_retries) -> None:
+        result["total_object_store_memory_gb"] = (
+            usage["object_store_memory"][1] * to_GiB
+        )
+    return result
+
+
+def get_cluster_config(cluster_config_file_path) -> dict:
+    """Get the static cluster (autoscaler) config used to launch this cluster."""
+
+    def get_instance_type(node_config):
+        if not node_config:
+            return None
+        if "InstanceType" in node_config:
+            # aws
+            return node_config["InstanceType"]
+        if "machineType" in node_config:
+            # gcp
+            return node_config["machineType"]
+        if (
+            "azure_arm_parameters" in node_config
+            and "vmSize" in node_config["azure_arm_parameters"]
+        ):
+            return node_config["azure_arm_parameters"]["vmSize"]
+        return None
+
+    try:
+        with open(cluster_config_file_path) as f:
+            config = yaml.safe_load(f)
+            result = {}
+            if "min_workers" in config:
+                result["min_workers"] = config["min_workers"]
+            if "max_workers" in config:
+                result["max_workers"] = config["max_workers"]
+
+            if "provider" in config and "type" in config["provider"]:
+                result["cloud_provider"] = config["provider"]["type"]
+
+            if "head_node_type" not in config:
+                return result
+            if "available_node_types" not in config:
+                return result
+            head_node_type = config["head_node_type"]
+            available_node_types = config["available_node_types"]
+            for available_node_type in available_node_types:
+                if available_node_type == head_node_type:
+                    head_node_instance_type = get_instance_type(
+                        available_node_types[available_node_type].get("node_config")
+                    )
+                    if head_node_instance_type:
+                        result["head_node_instance_type"] = head_node_instance_type
+                else:
+                    worker_node_instance_type = get_instance_type(
+                        available_node_types[available_node_type].get("node_config")
+                    )
+                    if worker_node_instance_type:
+                        result["worker_node_instance_types"] = result.get(
+                            "worker_node_instance_types", set()
+                        )
+                        result["worker_node_instance_types"].add(
+                            worker_node_instance_type
+                        )
+            if "worker_node_instance_types" in result:
+                result["worker_node_instance_types"] = list(
+                    result["worker_node_instance_types"]
+                )
+            return result
+    except Exception:
+        # If the file doesn't exist or the config file is invalid,
+        # then it's not a ray cluster or it's a k8s cluster.
```

Comment:
Shouldn't k8s still have a cluster yaml? Also probably catch a narrower class of exception? Shouldn't just `IOError` suffice here?

Reply:
> Also probably catch a narrower class of exception? Shouldn't just IOError suffice here?

It can also be `yaml.parser.ParserError` but I want to be extra safe here: getting cluster config is a best-effort thing, it shouldn't affect other code.

Reply:
I think we should come back to K8s later.

Reply:
OSS Ray operator keeps multiple cluster yamls somewhere other than the place where telemetry is running.
Nascent KubeRay support fetches cluster config info from a CRD.
Too complex to figure this stuff out now -- should come back to it when KubeRay is more solid as the main choice.

Reply:
hmmm k. i guess we don't have to do it now, but it does seem like we're solving quite a few of the same problems that `ray status`/load metrics summary already dealt with (handling types without names)

Reply:
What do you mean by `handling types without names`?